src/authV2/azext_authV2/custom.py

```diff
@@ -0,0 +1,836 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import json
+from knack.prompting import prompt_y_n
+from knack.util import CLIError
+from azure.cli.core.util import send_raw_request
+from azure.cli.command_modules.appservice._appservice_utils import _generic_site_operation
+from azure.cli.command_modules.appservice.custom import update_app_settings
+from azure.cli.core.commands.client_factory import get_subscription_id
+from azure.cli.command_modules.appservice._params import AUTH_TYPES
+
+MICROSOFT_SECRET_SETTING_NAME = "MICROSOFT_PROVIDER_AUTHENTICATION_SECRET"
+FACEBOOK_SECRET_SETTING_NAME = "FACEBOOK_PROVIDER_AUTHENTICATION_SECRET"
+GITHUB_SECRET_SETTING_NAME = "GITHUB_PROVIDER_AUTHENTICATION_SECRET"
+GOOGLE_SECRET_SETTING_NAME = "GOOGLE_PROVIDER_AUTHENTICATION_SECRET"
+MSA_SECRET_SETTING_NAME = "MSA_PROVIDER_AUTHENTICATION_SECRET"
+TWITTER_SECRET_SETTING_NAME = "TWITTER_PROVIDER_AUTHENTICATION_SECRET"
+
+
+# region rest calls
+
+
+def get_resource_id(cmd, resource_group_name, name, slot):
+    sub_id = get_subscription_id(cmd.cli_ctx)
+    resource_id = "/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Web/sites/{}".format(
+        sub_id,
+        resource_group_name,
+        name)
+    if slot is not None:
+        resource_id = resource_id + "/slots" + slot
+    return resource_id
+
+
+def get_auth_settings_v2(cmd, resource_group_name, name, slot=None):
+    resource_id = get_resource_id(cmd, resource_group_name, name, slot)
+    request_url = "https://management.azure.com/{}/{}?api-version={}".format(
```

Comment:
This means it will work on public cloud only. & throw exceptions when this runs on non public clouds. 
Alternately you can add a check to make these calls in Public cloud only - if the feature is not supported.

Reply:
This is supported in all clouds so made a change to dynamically read in the resource manager hostname from the cmd.cli_ctx